pyanaconda/payload/dnfpayload.py

```diff
@@ -1207,6 +1216,47 @@ def updateBaseRepo(self, fallback=True, checkmount=True):
                 if repo in enabled_repos:
                     self._fetch_md(repo)
 
+    def _getBaseRepoUrl(self, url):
+        """Try to find base repository from the treeinfo file."""
+        if self._treeinfo:
+            variants = self._treeinfo.variants
+
+            for variant in variants:
+                if variant in self.DEFAULT_REPOS:
+                    variant_obj = variants[variant]
```

Comment:
Please, iterate over keys and items to avoid this.

Reply:
Yeah I wanted to but GTK doesn't allow that :(

Reply:
Ok :-)